### PR TITLE
Thinking in React

### PR DESCRIPTION
**Pull request by arbob**
Thought process before building any react application. This article helped me a lot in thinking about the react way of doing things. Hope it helps other students too.
PR originally opened at the end of the course but as suggested by the TOP community members, it should be here.

 - [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).

#### 1.Describe the changes made:

...your text here

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number below:

#XXXXX

### CODE REVIEW

File: javascript/react-js/state-and-props.md [hunk 1]
```diff
@@ -232,6 +232,10 @@ For the state in functional components, it is a little bit different. Before the
 - How to handle state and props in a class component
 - Understanding setState()
 
+### Thinking in React
+
+An [article](https://reactjs.org/docs/thinking-in-react.html) by the React team on the thought process that should go into the developer's minds while building a react based apllication.
```
Comment (CatQueenCodes) [inline javascript/react-js/state-and-props.md:237]: Please capitalize the last "react" in the sentence.

**Review by CatQueenCodes — CHANGES_REQUESTED** [review]

File: javascript/react-js/state-and-props.md (hunk 1 above)
Comment (arbob) [reply javascript/react-js/state-and-props.md]: Done.

File: javascript/react-js/state-and-props.md [hunk 2]
```diff
@@ -232,6 +232,10 @@ For the state in functional components, it is a little bit different. Before the
 - How to handle state and props in a class component
 - Understanding setState()
 
+### Thinking in React
+
```
Comment (twalton83) [inline javascript/react-js/state-and-props.md:236]: I'm going to be honest, I feel like this part is sort of tacked on. This should instead be an assignment bullet point or an additional resource.

**Review by twalton83 — CHANGES_REQUESTED** [review]

**Comment by twalton83** [comment]
Also, quick thing @arbob, for your last few PRs you've been incorrectly filling out the template. Take a look at your original post closely, and edit it to match our template, and please be mindful moving forward of the formatting, thank you!

**Comment by arbob** [comment]
noted @twalton83 . Thanks for the reminder. Apologies.

File: javascript/react-js/state-and-props.md (hunk 2 above)
Comment (arbob) [reply javascript/react-js/state-and-props.md]: added.

**Comment by CatQueenCodes** [comment]
Thank you for the contribution!

**Pull request merged**